objectivec/test/ort_session_test.mm

```diff
@@ -0,0 +1,162 @@
+  NSString* testDataDir = [NSString stringWithFormat:@"%@/Contents/Resources/testdata",
+                                                     [[NSBundle bundleForClass:[ORTSessionTest class]] bundlePath]];
+  return [testDataDir stringByAppendingString:relativePath];
+}
+
+// model with an Add op
+// inputs: A, B
+// output: C = A + B
++ (NSString*)getAddModelPath {
+  return [ORTSessionTest getTestDataWithRelativePath:@"/single_add.onnx"];
+}
+
++ (NSMutableData*)dataWithScalarFloat:(float)value {
+  NSMutableData* data = [[NSMutableData alloc] initWithBytes:&value length:sizeof(value)];
+  return data;
+}
+
++ (ORTValue*)ortValueWithScalarFloatData:(NSMutableData*)data {
+  NSArray<NSNumber*>* shape = @[ @1 ];
+  NSError* err = nil;
+  ORTValue* ort_value = [[ORTValue alloc] initTensorWithData:data
+                                                 elementType:ORTTensorElementDataTypeFloat
+                                                       shape:shape
+                                                       error:&err];
+  XCTAssertNotNil(ort_value);
+  XCTAssertNil(err);
+  return ort_value;
+}
+
+- (void)testInitAndRunWithPreallocatedOutputOk {
+  NSMutableData* a_data = [ORTSessionTest dataWithScalarFloat:1.0f];
+  NSMutableData* b_data = [ORTSessionTest dataWithScalarFloat:2.0f];
+  NSMutableData* c_data = [ORTSessionTest dataWithScalarFloat:0.0f];
+
+  ORTValue* a = [ORTSessionTest ortValueWithScalarFloatData:a_data];
+  ORTValue* b = [ORTSessionTest ortValueWithScalarFloatData:b_data];
+  ORTValue* c = [ORTSessionTest ortValueWithScalarFloatData:c_data];
+
+  NSError* err = nil;
+  ORTSession* session = [[ORTSession alloc] initWithEnv:self.ortEnv
+                                              modelPath:[ORTSessionTest getAddModelPath]
+                                                  error:&err];
+  XCTAssertNotNil(session);
+  XCTAssertNil(err);
+
+  BOOL run_result = [session runWithInputs:@{@"A" : a, @"B" : b}
+                                   outputs:@{@"C" : c}
+                                     error:&err];
+  XCTAssertTrue(run_result);
+  XCTAssertNil(err);
+
+  const float c_expected = 3.0f;
+  float c_actual;
+  memcpy(&c_actual, c_data.bytes, sizeof(float));
+  XCTAssertEqual(c_actual, c_expected);
+}
+
+- (void)testInitAndRunOk {
+  NSMutableData* a_data = [ORTSessionTest dataWithScalarFloat:1.0f];
+  NSMutableData* b_data = [ORTSessionTest dataWithScalarFloat:2.0f];
+
+  ORTValue* a = [ORTSessionTest ortValueWithScalarFloatData:a_data];
+  ORTValue* b = [ORTSessionTest ortValueWithScalarFloatData:b_data];
+
+  NSError* err = nil;
+  ORTSession* session = [[ORTSession alloc] initWithEnv:self.ortEnv
+                                              modelPath:[ORTSessionTest getAddModelPath]
+                                                  error:&err];
+  XCTAssertNotNil(session);
+  XCTAssertNil(err);
+
+  NSDictionary<NSString*, ORTValue*>* outputs =
+      [session runWithInputs:@{@"A" : a, @"B" : b}
```

Comment:
Do 'outputs' automatically free themselves when they go out of scope or is a call to release the memory required somewhere?

Reply:
with automatic reference counting (ARC), outputs should automatically be freed.